### PR TITLE
i2s intrerrupt fix (IDFGH-9654)

### PR DESCRIPTION
**Pull request by dizcza**
Fixes https://github.com/espressif/esp-idf/issues/10994

### CODE REVIEW

**Comment by Alvin1Zhang** [comment]
Thanks for your contribution.

**Comment by dizcza** [comment]
Don't know why CI complains.

File: components/driver/i2s/i2s_std.c [hunk 1]
```diff
@@ -217,7 +217,7 @@ esp_err_t i2s_channel_init_std_mode(i2s_chan_handle_t handle, const i2s_std_conf
     }
 #endif
     ESP_GOTO_ON_ERROR(i2s_std_set_clock(handle, &std_cfg->clk_cfg), err, TAG, "initialize channel failed while setting clock");
-    ESP_GOTO_ON_ERROR(i2s_init_dma_intr(handle, ESP_INTR_FLAG_LEVEL1), err, TAG, "initialize dma interrupt failed");
+    ESP_GOTO_ON_ERROR(i2s_init_dma_intr(handle, I2S_INTR_ALLOC_FLAGS), err, TAG, "initialize dma interrupt failed");
```
Comment (L-KAYA) [inline components/driver/i2s/i2s_std.c:220]: Could you please update the corresponding lines in `i2s_pdm.c` and `i2s_tdm.c` as well?

File: components/driver/i2s/include/driver/i2s_common.h [hunk 2]
```diff
@@ -16,6 +16,18 @@
...
+// If ISR handler is allowed to run whilst cache is disabled,
+// Make sure all the code and related variables used by the handler are in the SRAM
+#if CONFIG_I2S_ISR_IRAM_SAFE
+#define I2S_INTR_ALLOC_FLAGS    (ESP_INTR_FLAG_IRAM | ESP_INTR_FLAG_INTRDISABLED | ESP_INTR_FLAG_SHARED | ESP_INTR_FLAG_LOWMED)
+#define I2S_MEM_ALLOC_CAPS      (MALLOC_CAP_INTERNAL | MALLOC_CAP_8BIT)
+#else
+#define I2S_INTR_ALLOC_FLAGS    (ESP_INTR_FLAG_INTRDISABLED | ESP_INTR_FLAG_SHARED | ESP_INTR_FLAG_LOWMED)
+#define I2S_MEM_ALLOC_CAPS      MALLOC_CAP_DEFAULT
+#endif //CONFIG_I2S_ISR_IRAM_SAFE
+#define I2S_DMA_ALLOC_CAPS      (MALLOC_CAP_INTERNAL | MALLOC_CAP_DMA)
```
Comment (L-KAYA) [inline components/driver/i2s/include/driver/i2s_common.h:29]: These macros are not supposed to be public, could you please move them into `i2s_private.h` instead?

**Review by L-KAYA — COMMENTED** [review]
@dizcza  Really thanks for your bugfix and apologize for the careless bug

**Comment by L-KAYA** [comment]
> Don't know why CI complains.

The Ci failed due to the copyright year is  not updated, you can install the `pre-commit hook` (see [how to install the pre-commit hook](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/contribute/install-pre-commit-hook.html)) before adding the commit, it will help to fix the copyright issue.

**Comment by dizcza** [comment]
> The Ci failed due to the copyright date is not updated, you can install the pre-commit hook (see [how to install the pre-commit hook](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/contribute/install-pre-commit-hook.html)) before adding the commit, it will help to fix the copyright issue.

that's exactly what I did in my latest commit

**Comment by L-KAYA** [comment]
> that's exactly what I did in my latest commit

To trigger the copyright check, you may need to do a soft reset and commit again.
BTW, could you please also squash the commit, the first commit seems change nothing, thanks!

**Pull request closed without merging**